PublicMapping/districtbuilder
# Export project GeoJSON

## Overview
Export GeoJSON for a given project

### Demo
Viewing the GeoJSON for a project using Delaware data

#### Delaware with all districts unassigned
![db_qgis_de_all_unassigned](https://user-images.githubusercontent.com/2926237/81883264-bdb04180-9562-11ea-87d5-69f5f7f26548.png)

#### Delaware with 3 districts
![db_qgis_de_123_assigned](https://user-images.githubusercontent.com/2926237/81883263-bdb04180-9562-11ea-856f-baca5682add5.png)

#### More complex map
![db_qgis_de_complex](https://user-images.githubusercontent.com/2926237/81884861-01a54580-9567-11ea-9492-363c450fe103.png)

## Testing Instructions
* Delete all projects since we've added a new non-nullable field to the `project` table
* Run migrations
* Ensure you have a region config set up that corresponds to the hardcoded Delaware topology (`s3_uri` must be 'de')
```
districtbuilder=# select * from region_config where s3_uri = 'de';
-[ RECORD 1 ]+-------------------------------------
id           | 9490b40c-6468-4269-aa63-18cccc2a7d47
name         | Delaware
country_code | US
region_code  | DE
s3_uri       | de
version      | 2020-05-13 21:35:34.223246+00
```
* Create a project through the UI that uses that region config and has 3 districts
* Access the endpoint to produce GeoJSON for that project
```
$ http :3003/api/projects/<project_id>/export/geojson 'Authorization:Bearer $TOKEN' > ~/Downloads/myGeoJSON.json
```
* Check that the GeoJSON looks as you'd expect in QGIS
* Modify the districts via PATCH like so:
```
$ http PATCH :3003/api/projects/<project_id> districtsDefinition:='[1, 2, [1,1,1,2,2,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]' Authorization:Bearer $TOKEN' > ~/Downloads/myOtherGeoJSON.json
```
* Check that the other GeoJSON looks as you'd expect in QGIS

Closes #92
